core/trino-main/src/main/java/io/trino/operator/join/unspilled/ArrayPositionLinks.java

```diff
@@ -0,0 +1,108 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.trino.operator.join.unspilled;
+
+import io.airlift.slice.Slices;
+import io.airlift.slice.XxHash64;
+import io.trino.operator.join.JoinFilterFunction;
+import io.trino.spi.Page;
+import org.openjdk.jol.info.ClassLayout;
+
+import java.util.Arrays;
+import java.util.List;
+
+import static io.airlift.slice.SizeOf.sizeOf;
+import static java.util.Objects.requireNonNull;
+
+public final class ArrayPositionLinks
```

Comment:
let's fork classes that are related to spill for now. We can always pull other related classes too